code/controllers/CMSPageHistoryController.php

```diff
@@ -358,7 +358,7 @@ public function ShowVersionForm($versionID = null) {
 	/**
 	 * @param int $versionID
 	 * @param int $otherVersionID
-	 * @return Form
+	 * @return mixed Form|boolean|SS_HTTPResponse
```

Comment:
`Security::permissionFailure()` does return a response, it’s `RequestHandler::httpError()` that doesn’t (this gets me every time!)